cmd/podman/create.go

```diff
@@ -300,13 +266,98 @@ func parseSecurityOpt(config *createConfig, securityOpts []string) error {
 	return err
 }
 
+func exposedPorts(c *cli.Context, imageExposedPorts map[string]struct{}) (map[nat.Port]struct{}, map[nat.Port][]nat.PortBinding, error) {
+	// TODO Handle exposed ports from image
+	// Currently ignoring imageExposedPorts
+
+	ports, portBindings, err := nat.ParsePortSpecs(c.StringSlice("publish"))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, e := range c.StringSlice("expose") {
+		// Merge in exposed ports to the map of published ports
+		if strings.Contains(e, ":") {
+			return nil, nil, fmt.Errorf("invalid port format for --expose: %s", e)
+		}
+		//support two formats for expose, original format <portnum>/[<proto>] or <startport-endport>/[<proto>]
+		proto, port := nat.SplitProtoPort(e)
+		//parse the start and end port and create a sequence of ports to expose
+		//if expose a port, the start and end port are the same
+		start, end, err := nat.ParsePortRange(port)
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid range format for --expose: %s, error: %s", e, err)
+		}
+		for i := start; i <= end; i++ {
+			p, err := nat.NewPort(proto, strconv.FormatUint(i, 10))
+			if err != nil {
+				return nil, nil, err
+			}
+			if _, exists := ports[p]; !exists {
+				ports[p] = struct{}{}
+			}
+		}
+	}
+	return ports, portBindings, nil
+}
+
+func imageData(c *cli.Context, runtime *libpod.Runtime, image string) (string, string, *libpod.ImageData, error) {
```

Comment:
Can you throw a comment on this function to explain what it returns

Reply:
Done